app/Rules/UpdateAssetReservationRule.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Rules;
+
+use App\Models\Reservation;
+use Carbon\Carbon;
+use Illuminate\Contracts\Validation\Rule;
+
+class UpdateAssetReservationRule implements Rule
+{
+
+    public $start_time;
+    public $end_time;
+    public $date;
+    public $id;
+    /**
+     * Create a new rule instance.
+     *
+     * @return void
+     */
+    public function __construct($date, $start_time, $end_time, $id)
+    {
+        $this->date = $date;
+        $this->start_time = $start_time;
+        $this->end_time = $end_time;
+        $this->id = $id;
+    }
+
+    /**
+     * Determine if the validation rule passes.
+     *
+     * @param  string  $attribute
+     * @param  mixed  $value
+     * @return bool
+     */
+    public function passes($attribute, $value)
+    {
+        $result = Reservation::whereTime('start_time', '>=', Carbon::parse($this->start_time))
+            ->whereTime('end_time', '<=', Carbon::parse($this->end_time))
+            ->where('date', Carbon::parse($this->date))
+            ->where($attribute, $value)->first();
```

Comment:
minor kang @rindibudiaramdhan , yang `->first()` kayaknya lebih baik ditaruh di baris berikutnya. CMIIW
```suggestion
            ->where($attribute, $value)
            ->first();
```

Reply:
baik kang @aldidoanta